extensions/interactions/guppy.html

```diff
@@ -1,33 +1,56 @@
-<link rel="stylesheet" href="/third_party/static/guppy-b5055b/build/guppy-default.min.css">
+<link rel="stylesheet" href="/third_party/static/guppy-175999/build/guppy-default.min.css">
 <script>
   // Use the normal default browser eventlistener without passive set to false.
   // TODO(#8833): Use passive listeners when guppy supports it.
   EventTarget.prototype.addEventListener = (
     EventTarget.prototype.addEventListener._original);
 </script>
-<script type="text/javascript" src="/third_party/static/guppy-b5055b/build/guppy.min.js"></script>
+<script type="text/javascript" src="/third_party/static/guppy-175999/build/guppy.min.js"></script>
 
 <script type="text/javascript">
   Guppy.init({
-    "path": "/third_party/static/guppy-b5055b/build"
+    'path': '/third_party/static/guppy-175999/build'
   });
+  Guppy.configure('buttons', ['controls']);
+  Guppy.configure('empty_content', '\\color{grey}{\\text{\\small{Type a formula here.}}}');
+
+  // Remove symbols since they are not supported.
+  var symbols_to_remove = ['norm', 'utf8', 'text', 'sym_name', 'eval', 'floor',
+    'factorial', 'sub', 'int', 'defi', 'deriv', 'sum', 'prod', 'root', 'vec',
+    'point', 'infinity', 'leq', 'less', 'geq', 'greater', 'neq'];
+  for(var symbol of symbols_to_remove) {
```

Comment:
Add space after "for".

I am also a little concerned about having all this in the plain JS. Could you please check with @srijanreddy98 and @bansalnitish whether there are any issues with this from an Angular perspective?

Reply:
@seanlip, @iamprayush there is an issue as this can't be used in angular2+. You can't use script tags in angular HTML. Angular will remove it. I am not really sure if my suggested method works in ajs but here is how this is done in angular2+:
Case: When you can't find an npm guppy module and have to use the script tag
In order to do this thing in angular, you will have to add the script tag using the component.ts file. Here is a generalized function to that.

```
  loadExternalScript(scriptUrl: string): Promise<boolean> {
    return new Promise((resolve, reject) => {
      const scriptElement = document.createElement('script');
      scriptElement.src = scriptUrl;
      scriptElement.onload = () => resolve(true);
      document.body.appendChild(scriptElement);
    });
  }
```
Declare the variable at the top of the file, right after imports:
```
declare let Guppy: any;
```

Then load the script and add all the contents of this script tag in `then()`.
```
loadExternalScript('/third_party/static/guppy-175999/build/guppy.min.js').then(
_ => { 
  Guppy.init({});
  ...
});
```

Reply:
@srijanreddy98 Thanks for such a clear explanation!
My question is -- how did it work till now, then? Other static third party libs (midijs, pencilcode, etc.) are also imported in this way. All these libs are being loaded by the `dependency_html.html` file. Isn't this how it's supposed to be?

One more thing, these configs are supposed to be the global configs for all instances of guppy across the codebase, so that we don't have to repeat all of this code in each directive that we use it in. If I'm not mistaken, the method you're proposing would require these configs to be written in every file that guppy is being imported in, right?

Reply:
I think it worked because of `AngularJs`.

Reply:
> My question is -- how did it work till now, then?

I mentioned that this won't work in angular2+, I didn't say that it won't work in ajs. As a matter of fact, this approach is valid in ajs.  It works in ajs because it doesn't sanitize HTML. This is not a problem now, and mostly won't be when migrating to angular2+ because I have worked on such cases before (PayPal's js-sdk has the same problem). So @iamprayush, I would reach out to you during the migration of this code to make sure that the migrated code works as expected. 

> One more thing, these configs are supposed to be the global configs for all instances of guppy across the codebase

This won't be a problem. We can initialize in those in the root component and store them in a service and use that variable later on in other directives

> I think it worked because of AngularJs.

@nishantwrp yes, ajs doesn't sanitize HTML.

Reply:
FYI I spoke to @srijanreddy98 about this. I think it is not a good idea to have manipulation of `script` tags in the JS. It can lead to all kinds of injection vulnerabilities.

I suggested finding a way to have this in a service that gets initialized once at the appropriate time. Would it be possible to do that? Our main files should have "script src" tags and nothing else (especially not random custom JS).